### DISCUSSION TITLE
how to get the query with poor performance

### DISCUSSION THREAD

**Discussion by njalan** [trinodb/trino#18189] [Q&A]
is there any way to get the query with poor performance. it is easy to get slow query from mysql. Do we have the same function to get the slow query or how do you identify this query's performance is not good. Is there any metrics?

**Comment by jiangzhx** [comment]
I don't think there's currently an open source tool that can directly solve this problem. But you can try writing an event listener to collect the execution information of these SQL statements, and then perform the statistics work based on that information.

https://trino.io/docs/current/develop/event-listener.html

**Comment by njalan** [comment]
Thanks for your reply. I already got the execution statistics. For one same query the statistics is different when the cluster workload is different.
